src/app/devices/deviceList/components/deviceListCell.spec.tsx

```diff
@@ -0,0 +1,48 @@
+/***********************************************************
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import 'jest';
+import * as React from 'react';
+import { Icon } from 'office-ui-fabric-react/lib/Icon';
+import { DeviceListCell, DeviceListCellProps } from './deviceListCell';
+import { mountWithLocalization, testSnapshot } from '../../../shared/utils/testHelpers';
+
+describe('components/devices/deviceListCell', () => {
+    const deviceListCellProps: DeviceListCellProps = {
+        connectionString: 'string',
+        device: {
+            authenticationType: 'sas',
+            cloudToDeviceMessageCount: '0',
+            connectionState: 'connected',
+            deviceId: 'testDeviceId',
+            iotEdge: false,
+            lastActivityTime: '0001-01-01T00:00:00Z',
+            status: 'Enabled',
+            statusUpdatedTime: '0001-01-01T00:00:00Z'
+        },
+        itemIndex: 1,
+    };
+
+    const getComponent = (overrides = {}) => {
+        const props = {
+            ...deviceListCellProps,
+            ...overrides,
+        };
+        return <DeviceListCell {...props} />;
+    };
+
+    it('matches snapshot while loading', () => {
+        testSnapshot(getComponent());
```

Comment:
Would be good to render additional snapshots with the various `context.t(ResourceKeys.deviceLists.noData)` branches being hit.